ssl/ssl_cert.cc

```diff
@@ -302,7 +304,7 @@ static int cert_set_chain_and_key(
   }
 
   for (size_t i = 0; i < num_certs; i++) {
-    if (!PushToStack(certs_sk.get(), UpRef(certs[i]))) {
+    if (!PushToStack(certs_sk.get(), UpRef(sk_CRYPTO_BUFFER_value(certs->get(), i)))) {
       return 0;
     }
   }
```

Comment:
We could use `sk_CRYPTO_BUFFER_deep_copy` here:
```
  UniquePtr<STACK_OF(CRYPTO_BUFFER)> certs_sk(sk_CRYPTO_BUFFER_deep_copy(
      certs->get(), buffer_up_ref, CRYPTO_BUFFER_free));
```

The copy function `buffer_up_ref` is defined above.  It's also needed [here](https://github.com/aws/aws-lc/blob/c9d48a636986c82cd122a0d6cd586191aa9afb2c/ssl/ssl_session.cc#L217-L222). Perhaps the copy function should be defined somewhere they could share?